### DISCUSSION TITLE
Support for customizable pre-configured changelog locations

### DISCUSSION THREAD

**Discussion by zeldigas** [renovatebot/renovate#8500] [Suggest an Idea]
While renovate [maintains ad-hoc changelog locations](https://github.com/renovatebot/renovate/blob/master/lib/datasource/metadata.ts) this does not look optimal from scalability perspective - if I have changelog with custom location (e.g. https://bitbucket.org/connect2id/nimbus-jose-jwt/src/master/CHANGELOG.txt) the only option I have is to send PR, wait for it to be merged and update renovate tool version.

What I think would be great in addition to this option is:
1. Provide ability to maintain custom collection of changelog customization in config
2. It would be great to have supports for presets so that such collections can be shared without need to bump renovate versions
3. While resolving changelog file it makes sense to take into account configured creds from config (maybe it works right now, not sure)

Configuration might look close to what is currently maintained manually
```json5
"manualChangelogUrls": {
  "pypi" : {
     "myInternalPackage": "https://git.example.org/PRJ/pythonPackage/changelog.txt",
     "customPublicPackage": "https://github.com/user/repo/changes.md"
  }
}
"hostRules": [
// optional configuration to access git.example.org if it is not available without creds
]
```

So the question is - was it discussed already? If not and idea seems fine, I can translate it to issue.

**Comment by rarkins** [comment]
1. If you are in control of the package then the best solution is to have a predictable CHANGELOG location
2. If it's someone else's public package then contributing such a minor PR (one line diff?) to benefit other users too does not seem unreasonable considering the benefits you are presumably gaining from this tool. Such a PR can be contributed through GitHub's web UI and is typically merged within hours
3. Because we try to cache changelogs as much as possible, we can't allow user-controlled URLs *in the app* anyway. It could even be used for trolling or worse.

**Reply by zeldigas** [reply]
> If you are in control of the package then the best solution is to have a predictable CHANGELOG location

Do I get it right that non-github locations potentially requiring credentials are supported now?

> Such a PR can be contributed through GitHub's web UI and is typically merged within hours

I see your point, but the downside of this is that it requires bump of renovate image. While it not super-hard, but compared to configuration change this usually requires more efforts.

Speaking about public package custom locations - while in better world most projects would use similar, easily machine-readable layout, sad truth is that it's not always so. This is especially not good if metadata format to include location of changelog does not exist (e.g. in java/maven artifacts there is no dedicated key that leads to relying only on heuristic and assumption based on developer connection). All this leads to unequal expirience when working with github-hosted project and something more isolated, like maven plugins.

I'd be happy to have this config option available as self-hosted only option available in config.js file, but if you see it as a wrong direction -- ok.
